pkg/virt-handler/virtwrap/api/converter.go

```diff
@@ -48,6 +49,51 @@ func Convert_v1_Disk_To_api_Disk(diskDevice *v1.Disk, disk *Disk) error {
 	return nil
 }
 
+func diskBusFromMachine(bus, machine string) string {
+	if bus != "" {
+		// use what it was explicitely selected
+		return bus
+	}
+	// catches: "q35", "pc-q35-*"
+	// see /path/to/qemu-kvm -machine help
+	if strings.HasPrefix(machine, "pc-q35") || strings.HasPrefix(machine, "q35") {
+		return "sata"
+	}
+	// safe fallback for x86_64, but very slow
+	return "ide"
+}
+
+func diskDeviceFromBusAndIndex(deviceName, bus string, index int) string {
+	if deviceName != "" {
```

Comment:
To make this really right, we would need do consider, that a person might set the deviceName on e.g. the third disk to "sda", which means we would then have "sda" on the first and the third disk.

Reply:
Yes, we would need to scan all the drives and to somehow mark device names already taken. It is not that hard, just another look + map, but I wonder if we need this complexity at all. I mean, *here* I'm having one hard time finding for one use case. Thoughts?

Reply:
You can also completely remove the device name from the cluster api and just do the deviceName logic on the node where you are doing it right now. @fabiand @davidvossel thoughts?

Reply:
I'm in favor of removing the whole deviceName thing from the public API.

To me it leads more to confusion (as it's depending on guest support) than helping.

Reply:
great, I'll make the code expose bus (which is totally fair and helpful IMO) instead.